plugin/trino-hive/src/test/java/io/trino/plugin/hive/fs/FileSystemDirectoryLister.java

```diff
@@ -13,19 +13,35 @@
  */
 package io.trino.plugin.hive.fs;
 
+import com.google.inject.Inject;
 import io.trino.filesystem.Location;
 import io.trino.filesystem.TrinoFileSystem;
+import io.trino.plugin.hive.HiveConfig;
+import io.trino.plugin.hive.S3StorageClassFilter;
 import io.trino.plugin.hive.metastore.Table;
 
 import java.io.IOException;
 
 public class FileSystemDirectoryLister
         implements DirectoryLister
 {
+    private final S3StorageClassFilter s3StorageClassFilter;
+
+    @Inject
+    public FileSystemDirectoryLister(HiveConfig hiveClientConfig)
+    {
+        this(hiveClientConfig.getS3StorageClassFilter());
+    }
+
+    public FileSystemDirectoryLister(S3StorageClassFilter s3StorageClassFilter)
+    {
+        this.s3StorageClassFilter = s3StorageClassFilter;
```

Comment:
`this.filterPredicate = requireNonNull(filterPredicate, "filterPredicate is null");`